sunkit_instruments/goes_xrs/goes_chianti_tem.py

```diff
@@ -0,0 +1,317 @@
+import numpy as np
+import pandas as pd
+from scipy import interpolate
+
+from astropy import units as u
+from astropy.io import fits
+from astropy.time import Time
+from sunpy import timeseries as ts
+from sunpy.data import manager
+from sunpy.time import parse_time
+from sunpy.util.exceptions import warn_user
+
+__all__ = ["calculate_temperature_em"]
+
+
+def calculate_temperature_em(goes_ts, abundance="coronal"):
+    """
+    This function calculates the isothermal temperature and
+    corresponding volume emission measure of the solar soft X-ray
+    emitting plasma observed by GOES/XRS.
+
+    These are calculated based on methods described in White et al. 2005 [1]_ (see notes)
+    for which the GOES fluxes and channel ratios are used together with look-up tables
+    of CHIANTI atomic models to estimate isothermal temperature and emission measure.
+    Technically speaking, the method interpolates on the channel flux ratio using
+    pre-calcuated tables for the fluxes at a series of temperatures for fixed emission
+    measure.
+
+    The method here is almost an exact replica of what is available in SSWIDL,
+    namely, goes_chianti_tem.pro, and it has been tested against that for consistency.
+
+    It now also works for the GOES-16 and -17 data, and for the re-processed netcdf
+    GOES/XRS files for GOES 13-15.
+
+    Also note that this has only been tested on the high resolutions 1s/2s/3s data of GOES/XRS.
+
+    Parameters
+    ----------
+    goes_ts : `~sunpy.timeseries.XRSTimeSeries`
+        The GOES/XRS timeseries containing the data of both the xrsa and xrsb channels (in units of W/m**2).
+    abundance: {``coronal`` | ``photospheric``}, optional
```

Comment:
```suggestion
    abundance: {"coronal", "photospheric"}, optional
```
I can never remember if this is correct.